File: orchestrator/src/escalate.ts
import { config } from "./config.js";

// Telegram delivery. Escalations reach the human on the same surface Hermes
// uses — text brief + an ElevenLabs voice note.
async function tg(method: string, body: Record<string, unknown>): Promise<boolean> {
  if (!config.telegram.enabled) return false;
  const res = await fetch(
    `https://api.telegram.org/bot${config.telegram.botToken}/${method}`,
    {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ chat_id: config.telegram.chatId, ...body }),
    },
  );
  if (!res.ok) console.error(`telegram ${method} failed`, res.status, await res.text());
  return res.ok;
}

export async function sendText(text: string): Promise<boolean> {
  return tg("sendMessage", { text, parse_mode: "Markdown", disable_web_page_preview: false });
}

// Send mp3 as an audio message (plays inline in Telegram).
export async function sendVoiceNote(audio: Buffer, caption: string): Promise<boolean> {
  if (!config.telegram.enabled) return false;
  const form = new FormData();
  form.append("chat_id", config.telegram.chatId);
  form.append("caption", caption.slice(0, 1000));
  form.append(
    "audio",
    new Blob([new Uint8Array(audio)], { type: "audio/mpeg" }),
    "intel-brief.mp3",
  );
  const res = await fetch(
    `https://api.telegram.org/bot${config.telegram.botToken}/sendAudio`,
    { method: "POST", body: form },
  );
  if (!res.ok) console.error("telegram sendAudio failed", res.status, await res.text());
  return res.ok;
}

File: agent/stalker/escalate.py
"""Telegram escalation delivery. Sends text alerts and voice-note briefs.
Best-effort: never raises, prints on error, no-ops when Telegram is disabled.
"""
from __future__ import annotations

import httpx

from .config import settings

_TIMEOUT = 30


def _base() -> str:
    return f"https://api.telegram.org/bot{settings.telegram_bot_token}"


def send_text(text: str) -> bool:
    """Send a Markdown text message to the configured chat. Returns success."""
    if not settings.telegram_enabled:
        return False
    try:
        resp = httpx.post(
            f"{_base()}/sendMessage",
            data={
                "chat_id": settings.telegram_chat_id,
                "text": text,
                "parse_mode": "Markdown",
            },
            timeout=_TIMEOUT,
        )
        resp.raise_for_status()
        return bool(resp.json().get("ok"))
    except Exception as e:  # noqa: BLE001
        print(f"[telegram] send_text failed: {e}")
        return False


def send_voice_note(audio: bytes, caption: str) -> bool:
    """Send an mp3 intel brief as a Telegram audio message. Returns success."""
    if not settings.telegram_enabled:
        return False
    try:
        resp = httpx.post(
            f"{_base()}/sendAudio",
            data={
                "chat_id": settings.telegram_chat_id,
                "caption": caption,
            },
            files={
                "audio": ("intel-brief.mp3", audio, "audio/mpeg"),
            },
            timeout=_TIMEOUT,
        )
        resp.raise_for_status()
        return bool(resp.json().get("ok"))
    except Exception as e:  # noqa: BLE001
        print(f"[telegram] send_voice_note failed: {e}")
        return False
